paper

```diff
@@ -0,0 +1,298 @@
+        printf '                        | this default, such as:\n'
+        printf '                        |     . ./paper setup example\n'
+        printf '                        | Which will allow you to run "example" instead.\n'
+  }
+
+  ###########################
+  ### CHECK PREREQUISITES ###
+  ###########################
+  local -r java_not_found_message="was not found - download the JDK by following https://paper.readthedocs.io/en/latest/java-update/index.html"
+  # TODO: Verify which ones are required
+  _paper_is_dep_available git
+  _paper_is_dep_available patch
+  _paper_is_dep_available mvn
+  _paper_is_dep_available curl
+  _paper_is_dep_available javac "${java_not_found_message}"
+  _paper_is_dep_available jar "${java_not_found_message}"
+
+  ###############
+  ### LETS GO ###
+  ###############
+  case "${1:-}" in
+    rb|rbp|rebuild|rbf|rbfull)
+      (
+        local target_task
+        case "${2:-}" in
+          ''|all) target_task=rebuildPatches;;
+          a|api) target_task=rebuildApiPatches;;
+          s|server) target_task=rebuildServerPatches;;
+          *) printf 'Unknown target - %s\nExpected one of: all api server\n' "${2:-}" >&2; return 1;;
+        esac
+        cd "${basedir}"
+        if [[ "${1}" == @(rbf|rbfull) ]]; then
+          printf 'Rebuilding patches without filter\n' >&2
+          ./gradlew "${target_task}" --filter-patches=false || return "${?}"
+        else
+          ./gradlew "${target_task}" || return "${?}"
+        fi
+      );;
+    p|patch)
+      (
+        local target_task
+        case "${2:-}" in
+          ''|all) target_task=applyPatches;;
+          a|api) target_task=applyApiPatches;;
+          s|server) target_task=applyServerPatches;;
+          *) printf 'Unknown target - %s\nExpected one of: all api server\n' "${2:-}" >&2; return 1;;
+        esac
+        cd "${basedir}"
+        ./gradlew "${target_task}" || return "${?}"
+      );;
+    j|jar)
+      (
+        cd "${basedir}"
+        ./gradlew applyPatches || return "${?}"
+        ./gradlew paperclipJar || return "${?}"
+      );;
+    b|build)
+      (
+      printf 'NOT IMPLEMENTED YET\n' >&2 # TODO
+      return 1
+      );;
+    i|install)
+      (
+        cd "${basedir}"
+        ./gradlew applyPatches || return "${?}"
+        ./gradlew publishToMavenLocal || return "${?}"
+        printf "Paper jar successfully built and installed to local repo\n"
+      );;
+    pc|paperclip)
+      (
+        cd "${basedir}"
+        ./gradlew paperclipJar || return "${?}"
+      );;
+    make)
+      (
+        if [[ "${2:-}" = bacon ]]; then
+          cd "${basedir}"
+          ./gradlew applyPatches || return "${?}"
+          ./gradlew paperclipJar || return "${?}"
+        fi
+      );;
+    m|mcdev)
+      (
+        cd "${basedir}"
+        ./gradlew makeMcDevSrc || return "${?}"
+      );;
```

Comment:
This should be removed. I will be removing the task as soon as I find a proper way to implement this.

As it currently is, due to how IntelliJ handles interfacing with Gradle, this task does nothing.

Reply:
Removed.